Azure Services/Azure Monitor/Agents/Migration Tools/DCR Config Generator/WorkspaceConfigToDCRMigrationTool.ps1

```diff
@@ -111,10 +137,16 @@ function Get-DCRArmTemplate
         [Parameter(Mandatory=$true)][string] $Location,
         [ValidateSet("Linux", "Windows")]
         [Parameter(Mandatory=$true)][string] $PlatformType,
-        [Parameter(Mandatory=$true)][string] $FolderPath
+        [Parameter(Mandatory=$true)][string] $FolderPath,
+        [Parameter(Mandatory=$true)][string] $SubscriptionId
     )
 
     $dcrJson = Get-DCRBaseJson -ResourceGroupName $ResourceGroupName -WorkspaceName $WorkspaceName -PlatformType $PlatformType
+    $dataCollectionEndpoint = GetOrCreate-DataCollectionEndpoint -SubscriptionId $SubscriptionId -ResourceGroupName $ResourceGroupName -WorkspaceName $WorkspaceName
```

Comment:
Does the customer incur an extra cost if we create a DCE for them even though they don't need it?
Or are there any other drawbacks of creating a DCE when custom logs are not defined?

Reply:
Is there a to delay the creation of a DCE till after everything checks out from Log Analytics workspace ?

Reply:
It's challenging without refactoring existing code. Because we need the DCE ID in the params, we need to create it while during the `DCRArmTemplate` generation phase

Reply:
Maybe DCE can be passed in?

Reply:
Thanks Victor and Komla, I made that change